hail/python/hail/expr/aggregators/aggregators.py

```diff
@@ -486,19 +486,54 @@ def all(condition) -> BooleanExpression:
     return count_where(~condition) == 0
 
 
-@typecheck(expr=expr_any)
-def counter(expr) -> DictExpression:
+@typecheck(expr=expr_any, weight=nullable(expr_any))
+def counter(expr, weight=None) -> DictExpression:
     """Count the occurrences of each unique record and return a dictionary.
 
     Examples
     --------
     Count the number of individuals for each unique `SEX` value:
 
-    >>> table1.aggregate(hl.agg.counter(table1.SEX))  # doctest: +NOTEST
+    >>> table1.aggregate(hl.agg.counter(table1.SEX))  # doctest: +SKIP
     {'M': 2L, 'F': 2L}
 
+    For each sample and gene, count the number of alternate alleles in each
```

Comment:
this example seems like it belongs in a how-to, not the counter docs. I'd prefer an example using `weight=` here generically.

Weight also needs to be documented.

Reply:
gene consequence example removed.